third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -575,18 +595,20 @@ impl Interpreter {
             } => {
                 gas_meter.charge_native_function(cost, Option::<std::iter::Empty<&Value>>::None)?;
 
-                // Load the module that contains this function regardless of the traversal context.
-                //
-                // This is just a precautionary step to make sure that caching status of the VM will not alter execution
-                // result in case framework code forgot to use LoadFunction result to load the modules into cache
-                // and charge properly.
-                resolver
-                    .loader()
-                    .load_module(&module_name, data_store, module_store)
-                    .map_err(|_| {
-                        PartialVMError::new(StatusCode::FUNCTION_RESOLUTION_FAILURE)
-                            .with_message(format!("Module {} doesn't exist", module_name))
-                    })?;
+                // Note(George): V2 loads function directly to avoid this case completely!
```

Comment:
is whatever the below was precautioning from no longer relevant? then maybe we can add one sentence to sketch that.

Reply:
Added a sentence, but yes: V1 uses cache which maybe empty, but then there is this assumption that module must be in module cache when we load a function. This seems pretty fragile, and in V2 we always fetch a module from module storage and cache on read.